src/shared/container-engine-lib.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+# container engine helper functions to handle both podman and docker commands
+
+CICD_TOOLS_COMMON_LOADED=${CICD_TOOLS_COMMON_LOADED:-1}
+
+if [ "$CICD_TOOLS_CONTAINER_ENGINE_LOADED" -eq 0 ]; then
+    return 0
+fi
+
+if [ "$CICD_TOOLS_COMMON_LOADED" -ne 0 ]; then
+    source "${CICD_TOOLS_WORKDIR}/src/shared/common.sh"
+fi
+
+echo "loading container engine"
+
+CONTAINER_ENGINE_CMD=''
+
+container_engine_cmd() {
+
+    if [ -z "$(get_container_engine_cmd)" ]; then
+        if ! set_container_engine_cmd; then
+            return 1
+        fi
+    fi
+
+    if [ "$(get_container_engine_cmd)" = "podman" ]; then
+        podman "$@"
+    else
+        docker "--config=${DOCKER_CONF}" "$@"
+    fi
+}
+
+get_container_engine_cmd() {
+    echo -n "$CONTAINER_ENGINE_CMD"
+}
+
+set_container_engine_cmd() {
+
+    if _configured_container_engine_available; then
+        CONTAINER_ENGINE_CMD="$PREFER_CONTAINER_ENGINE"
+    else
+        if container_engine_available 'podman'; then
+            CONTAINER_ENGINE_CMD='podman'
+        elif container_engine_available 'docker'; then
+            CONTAINER_ENGINE_CMD='docker'
+        else
+            echo "ERROR, no container engine found, please install either podman or docker first"
+            return 1
+        fi
+    fi
+
+    echo "Container engine selected: $CONTAINER_ENGINE_CMD"
+}
+
+_configured_container_engine_available() {
```

Comment:
Suggest renaming to `_preferred_container_engine_available`

Reply:
name updated, thanks for the suggestion!